src/Test.UnitTests.Sarif/CacheTests.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Sarif.UnitTests
+{
+    public class CacheTests
+    {
+        [Fact]
+        public void CacheBasics()
+        {
+            int buildCount = 0;
+            Cache<int, int> cache = new Cache<int, int>(
+                (key) => { buildCount++; return key; }, 
```

Comment:
>key [](start = 48, length = 3)

Perhaps your builder could return, say, `100 * key` so that you can distinguish keys from values -- that is, you can be sure your code isn't using a key in a place where it meant to use a value.

Reply:
Good idea.